.github/actions/runner-select/action.yml

```diff
@@ -67,11 +50,54 @@ runs:
         # <https://github.com/servo/servo/settings/variables/actions>
         if [ -n '${{ inputs.NO_SELF_HOSTED_RUNNERS }}' ]; then
           echo 'NO_SELF_HOSTED_RUNNERS is set!'
-          fall_back_to_github_hosted
+          echo "fall_back_to_github_hosted=true" | tee -a $GITHUB_OUTPUT
+          exit 0
         fi
 
         if [ '${{ inputs.force-github-hosted-runner }}' = true ]; then
           echo 'inputs.force-github-hosted-runner is set!'
+          echo "fall_back_to_github_hosted=true" | tee -a $GITHUB_OUTPUT
+          exit 0
+        fi
+
+        echo "fall_back_to_github_hosted=false" | tee -a $GITHUB_OUTPUT
+
+        artifact_path=$(mktemp)
+        echo "artifact_path=$artifact_path" | tee -a $GITHUB_OUTPUT
+
+        echo "unique_id=$unique_id" | tee -a "$artifact_path"
+        echo "self_hosted_image_name=$self_hosted_image_name" | tee -a "$artifact_path"
+        echo "qualified_repo=${{ github.repository }}" | tee -a "$artifact_path"
+        echo "run_id=${{ github.run_id }}" | tee -a "$artifact_path"
+
+    - name: Publish artifact with args
+      if: ${{ !fromJSON(steps.init.outputs.fall_back_to_github_hosted) }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: servo-ci-runners_${{ steps.init.outputs.unique_id }}
+        path: ${{ steps.init.outputs.artifact_path }}
+
+    - id: decision
+      name: Find a server and reserve a runner
+      shell: bash
+      run: |
+        github_hosted_runner_label='${{ inputs.github-hosted-runner-label }}'
+        self_hosted_image_name='${{ inputs.self-hosted-image-name }}'
+        fall_back_to_github_hosted='${{ steps.init.outputs.fall_back_to_github_hosted }}'
+        unique_id='${{ steps.init.outputs.unique_id }}'
+
+        set -euo pipefail
+
+        fall_back_to_github_hosted() {
```

Comment:
I think its confusing that this function and the variable in line 86 have the same name.

Reply:
fixed in 8b8db608c9e575500b8f95d725d0ec9378f686c1